scripts/setup_osx.sh

```diff
@@ -1,38 +1,248 @@
+    return 1 || return 0
 }
 
 
-echo "Installing system dependencies"
-echo "You will probably be asked for your sudo password."
 
-brew install swig sdl sdl_image sdl_mixer sdl_ttf portmidi openmpi
+# THE DEFAULTS INITIALIZATION - POSITIONALS
+_positionals=()
+# THE DEFAULTS INITIALIZATION - OPTIONALS
+_arg_set_envvar="off"
 
-# Make sure that we're under the directory of the project
-cd "$(dirname "$0")/.."
+print_help ()
+{
+  printf '%s\n' "Installer of garage for Linux."
+  printf 'Usage: %s [--(no-)set-envvar] ' "$0"
+  printf '[-h|--help] <mjkey-path>\n' "$0"
+  printf '\t%s\n' "<mjkey-path>: Path of the MuJoCo key"
+  printf '\t%s' "--set-envvar,--no-set-envvar: Set environment variables "
+  printf '%s\n' "required by garage in .bash_profile (off by default)"
+  printf '\t%s\n' "-h,--help: Prints help"
+}
 
-echo "Setting up MuJoCo..."
-/bin/bash ./scripts/setup_mujoco.sh
-if [[ "${?}" -ne 0 ]]; then
-  echo -e "\e[0;31mError: MuJoCo couldn't be set up\e[0m"
-  exit 1
+parse_commandline ()
+{
+  while test $# -gt 0
+  do
+    _key="$1"
+    case "$_key" in
+      --no-set-envvar|--set-envvar)
+        _arg_set_envvar="on"
+        test "${1:0:5}" = "--no-" && _arg_set_envvar="off"
+        ;;
+      -h|--help)
+        print_help
+        exit 0
+        ;;
+      -h*)
+        print_help
+        exit 0
+        ;;
+      *)
+        _positionals+=("$1")
+        ;;
+    esac
+    shift
+  done
+}
+
+
+handle_passed_args_count ()
+{
+  _required_args_string="'mjkey-path'"
+  error_msg="$(echo "FATAL ERROR: Not enough positional arguments - we" \
+    "require exactly 1 (namely: $_required_args_string), but got only" \
+    "${#_positionals[@]}.")"
+  test ${#_positionals[@]} -ge 1 || _PRINT_HELP=yes die "${error_msg}" 1
+  error_msg="$(echo "FATAL ERROR: There were spurious positional arguments" \
+    "--- we expect exactly 1 (namely: $_required_args_string), but got" \
+    "${#_positionals[@]} (the last one was: '${_positionals[*]: -1}').")"
+  test ${#_positionals[@]} -le 1 || _PRINT_HELP=yes die "${error_msg}" 1
+}
+
+assign_positional_args ()
+{
+  _positional_names=('_arg_mjkey_path' )
+
+  for (( ii = 0; ii < ${#_positionals[@]}; ii++))
+  do
+    error_msg="$(echo "Error during argument parsing, possibly an Argbash" \
+      "bug.")"
+    eval "${_positional_names[ii]}=\${_positionals[ii]}" || \
+      die "${error_msg}" 1
+  done
+}
+
+parse_commandline "$@"
+handle_passed_args_count
+assign_positional_args
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+
+# Utility functions
+print_error() {
+  printf "\e[0;31m%s\e[0m" "${1}"
+}
+
+print_warning() {
+  printf "\e[0;33m%s\e[0m" "${1}"
+}
+
+# Verify there's a file in the mjkey path
+test -f "${_arg_mjkey_path}" || _PRINT_HELP=yes die \
+  "The path ${_arg_mjkey_path} of the MuJoCo key is not valid." 1
+
+# Make sure that we're under the garage directory
+GARAGE_DIR="$(readlink -f $(dirname $0)/..)"
```

Comment:
I got this error while running the script:

```
readlink: illegal option -- f
usage: readlink [-n] [file ...]
```

Reply:
It seems this option is not available in OS X. Even though greadlink is available, I wouldn't like to download a new tool just for this little thing. I will find a way to work this out with more primitive tools so there's no compatibility issue.

Reply:
OS X is BSD but not Linux, so you can only really rely on `bash` primitives and POSIX commands.

Reply:
This trick has worked for me before https://stackoverflow.com/questions/3572030/bash-script-absolute-path-with-osx